api/features/feature_segments/tests/test_views.py

```diff
@@ -258,6 +344,35 @@ def test_get_feature_segment_by_uuid(
     assert json_response["uuid"] == str(feature_segment.uuid)
 
 
+def test_get_feature_segment_by_uuid_for_staff(
+    feature_segment,
+    project,
+    staff_client,
+    staff_user,
+    environment,
+    feature,
+    with_environment_permissions,
+):
+    # Given
+    url = reverse(
+        "api-v1:features:feature-segment-get-by-uuid", args=[feature_segment.uuid]
+    )
+    with_environment_permissions([UPDATE_FEATURE_STATE])
+    user_project_permission = UserProjectPermission.objects.create(
+        user=staff_user, project=project
+    )
+    user_project_permission.add_permission(VIEW_PROJECT)
```

Comment:
We could create a `with_project_permissions` fixture here too, right?

Reply:
Yep doing that now :)